coderd/externalauth/externalauth_test.go

```diff
@@ -260,6 +260,38 @@ func TestRefreshToken(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, updated.OAuthAccessToken, dbLink.OAuthAccessToken, "token is updated in the DB")
 	})
+
+	t.Run("WithExtra", func(t *testing.T) {
+		t.Parallel()
+
+		db := dbfake.New()
+		fake, config, link := setupOauth2Test(t, testConfig{
+			FakeIDPOpts: []oidctest.FakeIDPOpt{
+				oidctest.WithExtra(func(email string) map[string]interface{} {
+					return map[string]interface{}{
+						"authed_user": map[string]interface{}{
+							"access_token": "slack-user-token",
+						},
+					}
+				}),
+			},
```

Comment:
Pulling comment from https://github.com/coder/coder/pull/10151#discussion_r1350809793

Should we add the existing fields to the params of `WithExtra`? Maybe just pass the `token` in and let the function mutate the existing map.

I ask because it would be ideal to pass the actual access token:
```golang
oidctest.WithExtra(func(email string, token map[string]interface{}) {
	token["authed_user"] = map[string]interface{}{
		"access_token": token["access_token"],
	}
	// Completely nuke the original access token?
	token["access_token"] = "bogus_unused_token"
}),
```

This would make it valid to the fake OIDC IDP.

The thing is the `refresh` works because the refresh token is set correctly.

But the access-token is invalidly set to `slack-user-token`. We only check `access-token` on `/user-info`, so maybe it does not matter. But I could see it being a benefit to set the extra value to the correctly generated `access_token`.

Reply:
The access token isn't being overridden here which is why I disagree, it's just adding an extra property.